drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java

```diff
@@ -17,90 +17,131 @@
 package org.drools.workbench.screens.scenariosimulation.webapp.client.editor;
 
 import java.util.Collections;
-import java.util.function.Consumer;
+import java.util.List;
 
+import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
+import javax.inject.Named;
 
 import com.google.gwt.core.client.GWT;
-import org.drools.workbench.screens.scenariosimulation.client.editor.ScenarioMenuItem;
+import com.google.gwt.user.client.ui.IsWidget;
+import elemental2.dom.DomGlobal;
+import elemental2.promise.Promise;
+import jsinterop.annotations.JsFunction;
+import jsinterop.base.Js;
 import org.drools.workbench.screens.scenariosimulation.client.popup.FileUploadPopupPresenter;
 import org.drools.workbench.screens.scenariosimulation.client.resources.i18n.ScenarioSimulationEditorConstants;
+import org.drools.workbench.screens.scenariosimulation.kogito.client.editor.ScenarioSimulationEditorKogitoWrapper;
 import org.drools.workbench.screens.scenariosimulation.webapp.client.popup.ScenarioSimulationKogitoLoadingScesimPopupPresenter;
 import org.drools.workbench.screens.scenariosimulation.webapp.client.services.TestingVFSService;
 import org.gwtbootstrap3.client.ui.Popover;
 import org.jboss.errai.common.client.api.ErrorCallback;
-import org.kie.workbench.common.widgets.client.menu.FileMenuBuilder;
 import org.uberfire.backend.vfs.Path;
 import org.uberfire.backend.vfs.PathFactory;
-import org.uberfire.client.annotations.WorkbenchMenu;
-import org.uberfire.client.annotations.WorkbenchScreen;
-import org.uberfire.client.promise.Promises;
-import org.uberfire.lifecycle.OnStartup;
+import org.uberfire.client.mvp.AbstractActivity;
+import org.uberfire.client.mvp.EditorActivity;
 import org.uberfire.mvp.Command;
 import org.uberfire.mvp.PlaceRequest;
-import org.uberfire.mvp.impl.DefaultPlaceRequest;
-import org.uberfire.workbench.model.menu.Menus;
-
-import static org.drools.workbench.screens.scenariosimulation.webapp.client.editor.ScenarioSimulationEditorKogitoTestingScreen.IDENTIFIER;
+import org.uberfire.security.ResourceType;
+import org.uberfire.workbench.model.ActivityResourceType;
+import org.uberfire.workbench.model.bridge.Notification;
 
 @ApplicationScoped
-@WorkbenchScreen(identifier = IDENTIFIER)
-public class ScenarioSimulationEditorKogitoTestingScreen extends AbstractScenarioSimulationEditorKogitoScreen {
+@Named(DMNDiagramEditorActivity.ID)
+public class DMNDiagramEditorActivity extends AbstractActivity implements EditorActivity {
+    public static final String ID = "DMNDiagramEditor";
```

Comment:
```suggestion
    public static final String ID = "ScenarioSimulationEditor";
```
Please set `ScenarioSimulationEditor`as ID.

Reply:
Yeap, sure